src/id.rs

```diff
@@ -279,51 +280,46 @@ impl Display for P2pNode {
     }
 }
 
-impl PartialOrd for P2pNode {
-    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(self.cmp(other))
+#[derive(Clone, Serialize, Deserialize)]
```

Comment:
Wonder if to make it all consistent it would make sense to implement Serialize/Deserialize as well. Not sure.

Reply:
To be honest I'm not entirely sure myself either, but my feeling is that we should just use the generated one